core/src/OpenMPTarget/Kokkos_OpenMPTarget_Parallel_MDRange.hpp

```diff
@@ -144,15 +146,16 @@ class ParallelFor<FunctorType, Kokkos::MDRangePolicy<Traits...>,
       typename Policy::point_type offset, const FunctorType& functor,
       const Policy& policy) const {
 #ifdef KOKKOS_IMPL_MDRANGE_USE_NO_TILES
-    const int begin_0 = policy.m_lower[0];
-    const int begin_1 = policy.m_lower[1];
+    (void)offset;
+    const unsigned long begin_0 = policy.m_lower[0];
+    const unsigned long begin_1 = policy.m_lower[1];
 
-    const int end_0 = policy.m_upper[0];
-    const int end_1 = policy.m_upper[1];
+    const unsigned long end_0 = policy.m_upper[0];
+    const unsigned long end_1 = policy.m_upper[1];
 
 #pragma omp target teams distribute parallel for collapse(2) map(to : functor)
-    for (ptrdiff_t i0 = begin_0; i0 < end_0; i0++) {
-      for (ptrdiff_t i1 = begin_1; i1 < end_1; i1++) {
+    for (auto i0 = begin_0; i0 < end_0; i0++) {
```

Comment:
The warnings here are of the kind
```
/tmp/kokkos/core/src/OpenMPTarget/Kokkos_OpenMPTarget_Parallel_MDRange.hpp:157:5: error: comparison of integers of different signs: 'long' and 'const unsigned long' [-Werror,-Wsign-compare]
```
and they don't disappear even if using `unsigned int`.